lib/runtime/src/task_spec/arg_ref.h

```diff
@@ -9,8 +9,24 @@
 
 namespace FlexFlow {
 
+/** 
+ * \class ArgRefType
+ * \brief Enum class denoting the three types of ArgRefs
```

Comment:
Ideally try to raise the level of the docs--like, what does this add over the existing code below? For example, breaking down this `brief`:

> Enum class 

Obvious from the fact that the line of code starts with `enum class`

> denoting the three

Obvious from the fact that there are three values below

> types of ArgRefs

Obvious from the fact that this is named `ArgRefType`. 

If there isn't any reasonable doc to add because the code is self-explanatory, that's fine, but ideally don't add docs that are just pure reiterations of the code. 

Examples of helpful docs here would be: what are each of these enum values, and how do they correspond to what the system does as a whole? Where are they used?